types/circomlibjs/index.d.ts

```diff
@@ -0,0 +1,343 @@
+// Type definitions for circomlibjs 0.1
+// Project: https://github.com/iden3/circomlibjs
+// Definitions by: cedoor <https://github.com/cedoor>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export type Poseidon = (arr: any[], state?: any, nOut?: number) => any;
+
+export type Point = [any, any];
+
+export interface Signature {
+    R8: any;
+    S: any;
+}
+
+export class SMT {
+    constructor(
+        db: SMTMemDb,
+        root: any,
+        hash0: (left: any, right: any) => any,
+        hash1: (key: any, value: any) => any,
+        F: any,
+    );
+
+    delete(key: any): any;
+
+    find(key: any): any;
+
+    insert(key: any, value: any): any;
+
+    update(key: any, newValue: any): any;
+}
+
+export class SMTMemDb {
+    constructor(F: any);
+
+    get(key: any): any;
+
+    getRoot(): any;
+
+    multiDel(dels: any[]): void;
+
+    multiGet(keys: any[]): any;
+
+    multiIns(inserts: any[]): void;
+
+    setRoot(root: any): void;
+}
+
+export interface BabyJub {
```

Comment:
I implemented typed wrapper around BabyJub for my own usage. I figured out the following types:

```typescript
type BabyJubPoint = [Uint8Array, Uint8Array]

const Generator: BabyJubPoint = babyjub.Generator
const Base8: BabyJubPoint = babyjub.Base8

function mulPointEscalar(p: BabyJubPoint, scalar: bigint|number): BabyJubPoint {
    return babyjub.mulPointEscalar(p, scalar)
}

function inCurve(p: BabyJubPoint): boolean {
    return babyjub.inCurve(p)
}

function inSubgroup(p: BabyJubPoint): boolean {
    return babyjub.inSubgroup(p)
}

function addPoint(p1: BabyJubPoint, p2: BabyJubPoint): BabyJubPoint {
    return babyjub.addPoint(p1, p2)
}
```